### ISSUE TITLE
Spacy filenames error

### ISSUE THREAD

**Issue by Konfekt** [ludovicchabant/vim-gutentags#11]
Seems as if gutentags is suprised to see filenames with spaces under OpenSuse 13.2 (I recall from your blog entry that you only tested it under Mac and Windows though):
```
"~/.config/autokey/data/My Phrases/Dsg.txt" [noeol][mac] 1L, 21C
No location list
Error detected while processing function <SNR>192_setup_gutentags..<SNR>192_update_tags:
line   88:
E172: Only one file name allowed: chdir /home/konfekt/.config/autokey/data/My Phrases
Error detected while processing function <SNR>192_setup_gutentags:
line   45:
E171: Missing :endif
```

**Comment by Konfekt** [comment]
Probably a mere `fnameescape()` suffices.

**Comment by ludovicchabant** [comment]
Hi! Sorry for the trouble. Truth is I never tested this stuff with filenames with spaces in them! (I know it's a shame). I just tried on Windows and sure enough it doesn't work, although not with the same error you have (it just silently does nothing and keeps the tags file locked). It seems like a weird Batch script problem though, and when I look at the generated command line that Vim executes, it seems OK. I'll try it on Mac and Ubuntu when I get home.

**Comment by Konfekt** [comment]
Alright, thanks for looking into it. Here on Linux, 
```
fnameescape('/home/konfekt/.config/autokey/data/My Phrases') = '/home/konfekt/.config/autokey/data/My\ Phrases'
'''
and `chdir` accepts the escaped path just fine.

**Comment by ludovicchabant** [comment]
So I fixed some bugs when files have spaces in them over here: https://bitbucket.org/ludovicchabant/vim-gutentags/commits/all  (I have some issues with hg-git so it's not pushed to Github yet). I just realized there are some *different* problems when it's the *project root directory* that has space in it (as opposed to the files) so that's maybe what you're getting! I'll fix that next.

**Comment by Konfekt** [comment]
Yep that's true. I should have filed this under 'spacy file path' rather than 'file name' error. The files I openend had no spaces (it were those given above).

**Comment by ludovicchabant** [comment]
More bugs fixed over on Bitbucket (my hg-git problems are not fixed yet), but it turns out it's a bit more complicated.

I pass the full path to the tags file to `ctags` because the user could have specified a custom file with `:GutentagsGenerate`, but this means the tags file's path itself could have spaces in them. Even when correctly escaped, `ctags` chokes on it, at least for me on Mac. I get some errors like:

    sort:open failed: " blah/tags": no such file or directory

(with the error pointing to the part of the path after the space)

So even with the fixes, it will likely still not work. Thankfully you can get around the problem by putting your tags files in a cache folder away from view, which I think is preferable anyway. You can do this with the `g:gutentags_cache_dir` option.

If you _don't_ want to put tags files in a separate folder _and_ you want spaces in your paths, it will take a bigger code change, I think, because of this bug in `ctags`... unless I'm missing something obvious.

**Comment by Konfekt** [comment]
Hmm... the tag file is put in the project root dir, where the .git folder is, isn't it? In my case above, that would have been `~/.config`, So no spaces in this path.

So once your commits hit github, I give it a try and let you know if it worked. I hopefully never put spaces in my project root paths.

**Comment by ludovicchabant** [comment]
Should be fixed with dbfde870e502d66e3b4bc62842955af6c4d77d9f. Reopen the issue if it's not! Thanks!

**Comment by Konfekt** [comment]
Can confirm that a `PlugUpdate vim-gutentags` killed the issue. Good job. Thanks!
